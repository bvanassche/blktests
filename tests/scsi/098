#!/bin/bash
# SPDX-License-Identifier: GPL-3.0+
# Copyright (C) 2022 Google LLC

. tests/zbd/rc
. common/null_blk
. common/scsi_debug

DESCRIPTION="test F2FS write hint support"
QUICK=1

requires() {
	_have_fio
	_have_driver f2fs
	_have_program mkfs.f2fs
	_have_scsi_debug_group_number_stats
}

test() {
	echo "Running ${TEST_NAME}"

	local mount_dir="$TMPDIR/mnt"

	local scsi_debug_params=(
		delay=0
		dev_size_mb=1024
		sector_size=4096
	)
	_init_scsi_debug "${scsi_debug_params[@]}" &&
	local dev="/dev/${SCSI_DEBUG_DEVICES[0]}" fail &&
	ls -ld "${dev}" >>"${FULL}" &&
	mkfs.f2fs "${dev}" >>"${FULL}" 2>&1 &&
	mkdir -p "${mount_dir}" &&
	mount -t f2fs -o whint_mode=fs-based "${dev}" "${mount_dir}" &&
	local fio_args=(
		--size=1M
		--directory="${mount_dir}"
		--time_based
		--runtime=10
	) &&
	_run_fio_verify_io "${fio_args[@]}" >>"${FULL}" 2>&1 ||
	fail=true

	umount "${mount_dir}" >>"${FULL}" 2>&1
	head -n 999 /sys/bus/pseudo/drivers/scsi_debug/group_number_stats >> "${FULL}"
	while read -r group count; do
		if [ "$count" -gt 0 ]; then echo "$group"; fi
	done < /sys/bus/pseudo/drivers/scsi_debug/group_number_stats
	_exit_scsi_debug

	if [ -z "$fail" ]; then
		echo "Test complete"
	else
		echo "Test failed"
		return 1
	fi
}
