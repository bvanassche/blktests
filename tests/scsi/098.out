Running scsi/098
0
2
3
Test complete
